keps/sig-network/4962-network-topology-standard/README.md

```diff
@@ -0,0 +1,938 @@
+
+The scheduler will prioritize switches according to the order outlined above, providing a standardized approach for network-aware scheduling across a range of configurations.
+
+If provided, Network QoS Annotations can be used to refine and enhance the details of link performance, enabling more precise scheduling decisions.
+
+#### Example 1: network topology representation with reserved network types:
+
+Consider the following network topology:
+
+![Netwotk topology with reserved network types](./img/topo-reserved-labels.png)
+
+Let's examine node `vm12` as an example. This node is connected to NVSwitch `nvl10` and network switch `sw11`, which in turn is connected to switches `sw21` and `sw31`.
+In this case, the node labels would be:
+```yaml
+network.topology.kubernetes.io/accelerator: nvl10
+network.topology.kubernetes.io/block: sw11
+network.topology.kubernetes.io/datacenter: sw21
+network.topology.kubernetes.io/zone: sw31
+```
+
+If we have additional information such as latency and/or bandwidth between the node and the switches, it can be provided in an annotation:
+```yaml
+network.qos.kubernetes.io/switches: {
+   "nvl10": {
+      "latency": "2us",
+      "bandwidth": "100Gbps"
+   },
+   "sw11": {
+      "latency": "50us",
+      "bandwidth": "40Gbps"
+   },
+   "sw21": {
+      "latency": "500us",
+      "bandwidth": "20Gbps"
+   },
+  "sw31": {
+      "latency": "1ms",
+      "bandwidth": "10Gbps"
+   }
+}
+```
+
+#### Example 2: network topology representation with reserved network types:
+
+Consider the following network topology:
+
+![Netwotk topology with reserved network types](./img/topo-reserved-labels2.png)
+
+Let's examine node `vm31` as an example. This node is connected to the top-of-rack switch `sw13`, which in turn is connected to switches `sw22` and `sw31`.
+
+In this case, the node labels would be:
+
+```yaml
+network.topology.kubernetes.io/block: sw13
+network.topology.kubernetes.io/datacenter: sw22
+network.topology.kubernetes.io/zone: sw31
+```
+
+Similar to the previous example, the optional QoS metrics can be provided in an annotation:
+
+```yaml
+network.qos.kubernetes.io/switches: {
+   "sw13": {
+      "latency": "50us",
+      "bandwidth": "40Gbps"
+   },
+   "sw22": {
+      "latency": "500us",
+      "bandwidth": "20Gbps"
+   },
+   "sw31": {
+      "latency": "1ms",
+      "bandwidth": "10Gbps"
+   }
+}
+```
+
+### Extensibility and Future-Proofing
+
+This proposal is designed with extensibility in mind, enabling the use of custom network types. This ensures that the standard can adapt to future advancements in cluster networking without requiring significant overhauls.
+
+For custom network types, Network QoS Annotations are required, with distance being the minimum mandatory metric. Specifying latency and bandwidth is optional, but including them can offer a more detailed view of link performance, enabling more efficient scheduling decisions.
```

Comment:
IIUIC the use of custom network types means that environment that use them  may not be compatible with other environments, that practically removes all the benefits of standarization, another thing where I think weighted/distances model can help better

Reply:
this was already addressed.